From 11aeaa407efa3c9ec62bb5dc3a53f3d0354f2319 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Tue, 30 Jan 2018 14:29:22 +0800
Subject: [PATCH 3288/5242] MLK-17475-02 net: phy: tja110x: remove phy
 config_init in phy device probe()

commit  e01b2b1eb48d1751b7e67254c0c8db84e522bb80 from
https://source.codeaurora.org/external/imx/linux-imx.git

It is not necessary to config phy during phy device probe since it
will be called when do phy connect like below follow:
...
of_phy_connect()->
	phy_attach_direct()->
		phy_init_hw()->
			config_init()
...

Reviewed-by: Pandy Gao <pandy.gao@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/phy/tja110x.c |    5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/net/phy/tja110x.c b/drivers/net/phy/tja110x.c
index 84e5399..7e06244 100644
--- a/drivers/net/phy/tja110x.c
+++ b/drivers/net/phy/tja110x.c
@@ -228,11 +228,6 @@ static int nxp_probe(struct phy_device *phydev)
 
 	phydev->priv = nxp_specific;
 
-	/* init the phy. Should be called from phy framework but is not ? */
-	err = phydev->drv->config_init(phydev);
-	if (err < 0)
-		return err;
-
 	/* register sysfs files */
 	err = sysfs_create_group(&phydev->mdio.dev.kobj, &nxp_attribute_group);
 	if (err)
-- 
1.7.9.5

